docs/technical-reference/Testing Framework.md

```diff
@@ -0,0 +1,114 @@
+testcase will only run when running **all tests** and can no longer be executed in isolation.
+
+
+### NUnit
+
+### TestCases
+NUnit's TestCases is a pattern where some test is executed multiple time with varying input data (like xUnit's TestCase).
+These test methods accept one or more parameters and bear one of these attributes (on top of the `[Test]` attribute):
+ `TestCase` or `TestCaseSource`.
+
+There are some differences
+between `TestCase` (data is fixed at build time) and `TestCaseSource` (data is discovered at run time).
+
+### TestCase
+Basically, each `TestCase` is considered to be a specific test, and NUnit generates a name for each of those using the
+`ToString()` method for each parameter (e.g: `Test(1,2)` for parameter 1 and 2).
+Then each test is run separately and a test result is provided for each test.
+**Caveat**: if the `ToString()` method is not overriden, multiple test cases will have the same name 
+(eg: `Test(MyClass`). Then, there will be no way to distinguish between tests as well as properly associate test
+results.
+
+### TestCaseSource
+Each test case is reported during the discovery phase, and NUnit generates a name for each of those using the
+`ToString()` method for each parameter (e.g: `Test(1,2)` for parameter 1 and 2).
+Then each test is run separately and a test result is provided for each test.
+**Caveat**: if the `ToString()` method is not overriden, multiple test cases will have the same name 
+(eg: `Test(MyClass`). Then, there will be no way to distinguish between tests as well as properly associate test
+results.
+When this happens, these appear as a single test resulting in multiple outcomes in Visual Studio.
+
+Here is a summarized timeline of tests execution:
+```
+   NUnit runner calls the TestCaseSource and gets every test case.
+TestCase start event
+   NUnit run test with first set of data
+   NUnit reports test reult
+TestCase end event
+TestCase start event
+   NUnit run test with second set of data
+   NUnit reports test reult
+TestCase end event
+   ...
+   NUnit reports lasy test reult
+TestCase end event
+TestCase start event
+   ...
+```
+
+The problems for Stryker are:
+1. If test cases have the same name, we cannot distinguish between test cases, so it must be handled as a single test. This *single test* reports multiple test cases so Stryker needs to wait either for one failure or for each testcase
+to succeed to establish the correct status. The good news is that discovery should provide the number of results to wait for.
+2. Every mutation encountered by the `TestCaseSource` method will be reported as covered by the first test case,
+whereas it is likely these mutations would need to be adequately associated with their realtive test case.
+3. TestCaseSource is called **before** Testcase start event which means Stryker may set the active mutation too late. 
+The only way to handle this is to force any concerned mutants to run in dedicated test sessions.
+
```

Comment:
I have not looked into MsTest, but
MsTest is not the most use framework and I assume there is a good integration between MsTest and VsTest. In any case, the remediation strategies I used to work around those issues should also work for MsTest based assemblies.
And, I am not sure this document is of any use, beyond technical curiosity. 
Completing means adding a new test project using MsTest and intense debugging to understand how it works.
I think It can wait for a rainy day :-).